File: baekjoon/class3/js/11047_동전0.js
const input = [];
const strToNumArr = (str) =>
  str.split(" ").map((numString) => Number(numString));

require("readline")
  .createInterface(process.stdin, process.stdout)
  .on("line", function (line) {
    input.push(line.trim());
  })
  .on("close", function () {
      const [n, k] = strToNumArr(input.shift());
      const 
  });

File: baekjoon/class3/11047_동전0.py
import sys

input = sys.stdin.readline

n, k = map(int, input().split())

coins = [int(input()) for _ in range(n)]

coins.sort(reverse=True)
cnt = 0

for i in coins:
    if k == 0:
        break
    cnt += k // i
    k %= i

print(cnt)
